docs/testing-guide.md

```diff
@@ -0,0 +1,362 @@
+3. **Simplicity over coverage** - A few clear tests are better than many brittle ones
+4. **Consistency with existing patterns** - Follow the style established in the codebase
+
+## What to Test
+
+### ✅ High-Value Tests
+
+**Public API and Behavior**
+```php
+// GOOD: Tests actual behavior users/developers experience
+public function test_module_blocks_are_registered(): void
+{
+    $this->assertTrue(
+        WP_Block_Type_Registry::get_instance()->is_registered('sitchco/test-block'),
+        'Blocks from module should be auto-registered via manifest'
+    );
+}
+```
+
+**Integration Points**
+```php
+// GOOD: Tests how components work together
+function test_active_module_initialization()
+{
+    $ModuleInstance = $this->container->get(ModuleTester::class);
+    $this->assertTrue($ModuleInstance->featureOneRan);
+}
+```
+
+**Utility Functions with Complex Logic**
+```php
+// GOOD: Tests reusable utilities with edge cases
+public function test_remember_option_with_ttl_respects_expiration_metadata(): void
+{
+    $result = Cache::rememberOption('key', fn() => 'value-one', 60);
+    $stored = get_option('key');
+    $this->assertArrayHasKey('__cache_meta', $stored);
+}
+```
+
+**Data Transformations**
+```php
+// GOOD: Tests that data flows correctly through the system
+public function testCollectionWrapsPostQueryCorrectly()
+{
+    $collection = new Collection($post_query);
+    $this->assertEquals($post_query->to_array(), $collection->to_array());
+}
+```
+
+### ❌ Low-Value Tests to Avoid
+
+**Infrastructure Implementation Details**
+```php
+// BAD: Tests internal mechanism, not behavior
+public function test_manifest_generates_hash(): void
+{
+    $manifest = $this->generator->generate($path);
+    $this->assertArrayHasKey('hash', $manifest);
+}
+```
+*Why avoid?* If the hash algorithm changes but blocks still register correctly, this test breaks unnecessarily.
+
+**Bootstrap and Container Setup**
+```php
+// BAD: Duplicates what integration tests already verify
+public function test_bootstrap_sets_global_container(): void
+{
+    $bootstrap->initialize();
+    $this->assertNotNull($GLOBALS['SitchcoContainer']);
+}
+```
+*Why avoid?* If the container doesn't work, every integration test fails. No need to test it explicitly.
+
+**Trivial Getters/Setters**
+```php
+// BAD: No logic to test
+public function test_get_name_returns_name(): void
+{
+    $obj->setName('Test');
+    $this->assertEquals('Test', $obj->getName());
+}
+```
+*Why avoid?* These break when you refactor to public properties. Test behavior that matters instead.
+
+**Cache Implementation Details**
+```php
+// BAD: Tests internal cache mechanics
+public function test_cache_cleared_on_manifest_regenerate(): void
+{
+    $registry->ensureFreshManifests();
+    $this->assertFalse(wp_cache_get('sitchco_blocks_manifest', 'sitchco'));
+}
+```
+*Why avoid?* Caching is an optimization detail. If blocks work correctly, the cache works correctly.
```

Comment:
Not sure I entirely agree with this recommendation. Because very few WP tests can be written as pure unit tests with easily testable inputs and outputs, one of the only ways to test complete behavior of a feature within a WP context is through side effects, i.e. changes to the database, filesystem, etc. If writes or deletes to object cache is important to the behavior of a feature, it can and should be verified. By doing this, you aren't really testing internal cache mechanics, you're testing the usage of collaborating WP services.